### PR TITLE
feat(claim): Add informative gas modal

### PR DESCRIPTION
**Pull request by niconiahi**
[POAP-400](https://xivis.atlassian.net/browse/POAP-400)

claim-modal

### CODE REVIEW

File: client/src/CodeClaimPage/QRHashForm.tsx [hunk 1]
```diff
@@ -69,6 +74,26 @@ const QRHashForm: React.FC<{error: boolean, loading: boolean, checkClaim: (hash:
         </Formik>
       </div>
       <ClaimFooterMessage />
+
+      <ReactModal isOpen={isModalOpen} shouldFocusAfterRender={true}>
```
Comment (lndgalante) [inline client/src/CodeClaimPage/QRHashForm.tsx:78]: Please prefer implicit prop boolean over explicit

```suggestion
      <ReactModal isOpen={isModalOpen} shouldFocusAfterRender>
```

As a side note if you find an ESLint rule for this to add would be great 👍

File: client/src/CodeClaimPage/QRHashForm.tsx [hunk 2]
```diff
@@ -69,6 +74,26 @@ const QRHashForm: React.FC<{error: boolean, loading: boolean, checkClaim: (hash:
...
+        <div className="admin-list-modal">
+          <div className='claim-modal-text-container'>
+            <p>The current surge in gas prices makes a POAP minting have a cost of between $2 and $6 in mining fees.</p>
+            <p>We are working hard on a scaling solution that should be ready in 6-8 weeks. </p>
+            <p>In the meantime please input your address in the field below and click on claim. 
+              That’s going to block this POAP to be minted for free once our new deployment is ready. 
+              Afterwards you can continue the process at your expense or just close this page. 
+              For learning more about what our plans are visit our discord. 
+            </p>
+            <p><a href="http://poap.xyz/discord" target="_blank" rel="noopener noreferrer">http://poap.xyz/discord</a></p>
```
Comment (lndgalante) [inline client/src/CodeClaimPage/QRHashForm.tsx:88]: Prefer HTTPS link over HTTP due to performance and security reasons

File: client/src/CodeClaimPage/QRHashForm.tsx [hunk 3]
```diff
@@ -69,6 +74,26 @@ const QRHashForm: React.FC<{error: boolean, loading: boolean, checkClaim: (hash:
...
+          </div>
+          <div className="claim-modal-cancel-container">
+            <div onClick={() => handleModal()} className={'close-modal'}>
```
Comment (lndgalante) [inline client/src/CodeClaimPage/QRHashForm.tsx:91]: Prefer avoiding the double arrow functions, this is what you are writing:

```javascript
            <div onClick={() => (() => setIsModalOpen(false))()} className={'close-modal'}>
```

This is what we want to write
```javascript
            <div onClick={handleModal} className={'close-modal'}>
```

And it translates to
```javascript
            <div onClick={() => setIsModalOpen(false)} className={'close-modal'}>
```

**Review by lndgalante — CHANGES_REQUESTED** [review]
Good work!

**Comment by lndgalante** [comment]
@ragonzal can we avoid hardcoding this message? I mean every time the mining fee changes we will have to change it manually, maybe we can investigate an API for it. Also, I guess our client will appreciate a long-term solution.

```javascript
  <p>The current surge in gas prices makes a POAP minting have a cost of between $2 and $6 in mining fees.</p>
```

File: client/src/CodeClaimPage/QRHashForm.tsx (hunk 1 above)
Comment (niconiahi) [reply client/src/CodeClaimPage/QRHashForm.tsx:78]: Screen Shot 2020-07-28 at 15 50 35

done

**Comment by ragonzal** [comment]
Thanks @niconiahi we will discard this branch

**Pull request closed without merging**